### DISCUSSION TITLE
Milvus Proxy is not ready yet. please wait

### DISCUSSION THREAD

**Discussion by KinjalViradiya** [milvus-io/milvus#19030] [Q&A and General discussion]
<_MultiThreadedRendezvous of RPC that terminated with:
	status = StatusCode.UNAVAILABLE
	details = "internal: Milvus Proxy is not ready yet. please wait"
	debug_error_string = "{"created":"@1662377973.845959094","description":"Error received from peer ipv6:[::1]:19530","file":"src/core/lib/surface/call.cc","file_line":966,"grpc_message":"internal: Milvus Proxy is not ready yet. please wait","grpc_status":14}"

I am able to connect to milvus but when i do :

from pymilvus import utility
utility.has_collection(collection_name)

i get above mentioned error? can someone please help?

**Comment by yhmo** [comment]
If the milvus server started successfully, the log of Proxy should have these lines:
```
[2022/09/06 10:01:47.162 +08:00] [DEBUG] [proxy/service.go:299] ["start Proxy server done"]
[2022/09/06 10:01:47.162 +08:00] [INFO] [components/proxy.go:54] ["Proxy successfully started"]
```

The error "Milvus Proxy is not ready yet" indicated the proxy is not ready, or its dependencies are not ready. The most possible reason is the rootcoord is not ready.  Check the log of rootcoord to see what happened.
